### PR TITLE
Upgrade realm-core to v13.17.2

### PR DESCRIPTION
**Pull request by nielsenko**
- Update realm-core to v13.17.2
- Update CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,16 +11,20 @@
...
-* Fixed a realm generator issue, when used in concert with MobX. ([%1372](https://github.com/realm/realm-dart/pull/1372))
+* Fixed a realm generator issue, when used in concert with MobX. ([#1372](https://github.com/realm/realm-dart/pull/1372))
+* Fix failed assertion for unknown app server errors (Core upgrade, since v12.9.0).
+* Testing the size of a collection of links against zero would sometimes fail (sometimes = "difficult to explain"). (Core upgrade, since v13.15.1)
+* Rare corruption of files on streaming format (often following compact, convert or copying to a new file). (Core upgrade, since v12.12.0)
+* Trying to search a full-text indexes created as a result of an additive schema change (i.e. applying the differences between the local schema and a synchronized realm's schema) could have resulted in an IllegalOperation error with the error code `Column has no fulltext index`. (Core upgrade, since v13.2.0).
+* Sync progress for DOWNLOAD messages from server state was updated wrongly. This may have resulted in an extra round-trip to the server. (Core upgrade, since v12.9.0)
```
Comment (desistefanova) [inline CHANGELOG.md:21]: These three items seem to be repeated. They are added above already from the upgrade to 13.17.1.

**Review by desistefanova — COMMENTED** [review]
The CHANGELOG needs to be fixed.

File: test/realm_object_test.dart [hunk 2]
```diff
@@ -414,32 +414,23 @@ Future<void> main([List<String>? args]) async {
...
+    } else {
       expect(json, contains('"$propertyName":"${date.toCoreTimestampString()}"'));
     }
   }
 
   final dates = [
-    DateTime.utc(1970).add(Duration(days: 100000000)),
-    DateTime.utc(1970).subtract(Duration(days: 99999999)),
+    // BUG: core timestamp serialization does not support years
+    // very far from 1970 currently currently
```
Comment (desistefanova) [inline test/realm_object_test.dart:427]:
```suggestion
    // very far from 1970 currently
```

Comment (desistefanova) [reply test/realm_object_test.dart:427]: Is this a bug or by purpose?

**Review by desistefanova — COMMENTED** [review]
The CHANGELOG needs to be fixed.

**Review by blagoev — APPROVED** [review]
looks good to me, with what @desistefanova commented

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,16 +11,20 @@
 * Rare corruption of files on streaming format (often following compact, convert or copying to a new file). (Core upgrade, since v12.12.0)
 * Trying to search a full-text indexes created as a result of an additive schema change (i.e. applying the differences between the local schema and a synchronized realm's schema) could have resulted in an IllegalOperation error with the error code `Column has no fulltext index`. (Core upgrade, since v13.2.0).
 * Sync progress for DOWNLOAD messages from server state was updated wrongly. This may have resulted in an extra round-trip to the server. (Core upgrade, since v12.9.0)
-* Sync errors included the error message twice (Core upgrade, since v13.16.0).
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply CHANGELOG.md:14]: Yes. We never released a version with v13.16.0

File: CHANGELOG.md (hunk 1 above)
Comment (nielsenko) [reply CHANGELOG.md]: Fixed

File: test/realm_object_test.dart (hunk 2 above)
Comment (nielsenko) [reply test/realm_object_test.dart:427]: I believe it is a bug. Realm represent timestamps as 64 bit seconds offset from epoch (1970) and 32 bit nano-seconds. Hence realm can represent offsets from epoch up to 9223372036854689408 days, yet realm-core cannot serialise that.

Comment (nielsenko) [reply test/realm_object_test.dart]: I elaborated on the comment to make it more precise. The reason the PR is still in draft was mostly to raise this issue with core first.

Comment (nielsenko) [reply test/realm_object_test.dart]: I have raised https://github.com/realm/realm-core/issues/6892

**Review by desistefanova — APPROVED** [review]

**Pull request merged**